submariner/IPSec-transport-mode.md

```diff
@@ -0,0 +1,114 @@
+
+# Cable Driver Enhancement - Overlay over IPSec Transport Mode
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline the enhancements needed for the Submariner Cable Driver to
+enable Overlay-over-IPSec in transport mode.
+
+For Overlay-over-IPSec in transport mode with further configuration, the benefits fall into the
+following categories:
+
+* **Performance**: Taking advantage of an overlay protocol wrapped in offloaded IPSec in transport mode can
+yield significant performance improvements. The existing IPSEC configuration performance on a pair of modern
+servers ~ 2Gbit maximum. Switching to IP GRE Tunnel protected by an offloaded transport mode IPSEC external
+“wrap” - 25Gbit. A difference of 12.5 times. Or in the case of VXLAN offloaded with an IPSec transport wrap - 9Gbit.
+* **Scale**: While it is theoretically possible to scale a system built around tunnel mode IPSEC to hundreds of nodes,
```

Comment:
Regarding scale, in order to use dynamic routing and QoS services we need to develop a full (complicated ) SW stack for these features.
Having the overlay layer is just one piece of the puzzle. IMO it isn't a significant advantage here.

Reply:
It is not an advantage. It is a required prerequisite. Dynamic routing and tunnel mode are two incompatible paradigms.

Reply:
I'm not sure how much L2/L3 VPN capabilities (like dynamic routing) are needed in SM context.
But, yeah using ipsec in transport mode enables the usage of L2/L3 VPN capabilities.